### PR TITLE
(0.41) Introduce JVMPortableRestoreMode

### PR DESCRIPTION
**Pull request by tajila**
Previously, the CRIURestoreNonPortableMode was used to determine if a checkpoint could be taken after restore. If CRIURestoreNonPortableMode was enabled (default mode) only a single checkpoint is allowed and the JVM doesn't need to generate portable code upon restore. This capability also allows the JVM to use standard security providers since there is no risk for that state to be serialized.

There is a need to separate out the portability capabilites and the behavioural changes that CRIURestoreNonPortableMode offers. There are cases where one wants portability as well as standard security capabilites for debugging purposes.

This PR separates out the portability capabilites and the behavioural aspects by providing an option that forces portability upon restore regardless of what CRIURestoreNonPortableMode is set to.

In the future we can add another option that forces portability in non-CRIU mode, it its desired.

Backport of https://github.com/eclipse-openj9/openj9/pull/18252

### CODE REVIEW

**Comment by tajila** [comment]
@ymanton Please add a comment once you've confirmed this addresses the issue

File: runtime/compiler/net/CommunicationStream.hpp [hunk 1]
```diff
@@ -118,7 +118,11 @@ class CommunicationStream
    // likely to lose an increment when merging/rebasing/etc.
    //
    static const uint8_t MAJOR_NUMBER = 1;
+<<<<<<< HEAD
```
Comment (pshipton) [inline runtime/compiler/net/CommunicationStream.hpp:121]: @TobiAjila this isn't a clean backport.

**Comment by ymanton** [comment]
> @ymanton Please add a comment once you've confirmed this addresses the issue

Confirmed, `-XX:+JVMPortableRestoreMode` appears to fix the issue I was seeing on ppc64le when taking a checkpoint on Power 10 and restoring on Power 9.

**Comment by tajila** [comment]
Thanks Younes

@pshipton this can be merged now

**Review by pshipton — APPROVED** [review]

**Pull request merged**